ext/pdo/pdo_stmt.c

```diff
@@ -2348,7 +2348,7 @@ PDO_API void php_pdo_free_statement(pdo_stmt_t *stmt)
 	do_fetch_opt_finish(stmt, 1);
 
 	if (!Z_ISUNDEF(stmt->database_object_handle)) {
-		zval_ptr_dtor(&stmt->database_object_handle);
+		Z_DELREF(stmt->database_object_handle);
```

Comment:
According to ZendMM and valgrind (running the test suite), it does not. Furthermore, searching for `database_object_handle` in `ext/pdo/pdo_dbh.c` shows that it's always handled with `Z_ADDREF`/`Z_DELREF`.

AIUI, `Z_DELREF` decreases the refcount and destructs the object if the refcount became zero., what would be the desired behavior. Please correct me, if I'm wrong. :-)